workspace: add READMEs (#3644)
dusk-network/rusk: contracts/transfer/README.md

```rust
pub fn withdraw(&mut self, withdraw: Withdraw) -> ()
```

### `convert`
> Can only be called by the transfer contract

**Description**: Performs an atomic conversion transacton between Phoenix notes and Moonlight balance using the deposit sent to the contract. The conversion transforms the type without changing the value.

```rust
pub fn convert(&mut self, convert: Withdraw) -> ()
```

### `contract_to_contract`

**Description**: Transfers funds from one contract's balance to another and calls a function on the receiving contract (specified in the transfer argument). Contracts receiving funds are expected to expose the function specified by the sender, which is called using a `ReceiveFromContract` as argument.

```rust
pub fn contract_to_contract(&mut self, transfer: ContractToContract) -> ()
```

### `contract_to_account`

**Description**: Transfers funds from a contract balance to a moonlight account.

```rust
pub fn contract_to_account(&mut self, transfer: ContractToAccount) -> ()
```

### `root`

> **Description**: Returns the current root of the tree as a cryptographic commitment to the current state. The root is essential for verifying note inclusion in the merkle tree and validating phoenix transactions.

Reviewer: ```suggestion
**Description**: Returns the current root of the merkle tree of all phoenix-notes as a cryptographic commitment to the current state. The root is essential for verifying note inclusion in the tree and validating phoenix transactions.
```

```rust
pub fn root(&self) -> BlsScalar
```

### `account`

**Description**: Retrieves the moonlight account data i.e., balance and nonce that is associated with the specified public key.

```rust
pub fn account(&self, key: &AccountPublicKey) -> AccountData
```

### `contract_balance`

**Description**: Returns the balance of the specified contract.

```rust
pub fn contract_balance(&self, contract_id: &ContractId) -> u64
```

### `opening`

**Description**: Retrieves the merkle opening for a note-hash at the specified position. Returns None if the position is invalid or the note doesn't exist.

```rust
pub fn opening(&self, pos: u64) -> Option<NoteOpening>
```

### `existing_nullifiers`

> **Description**: Based on a given list of nullifiers, returns only the nullifiers that already exist in the contract. Nullifiers "nullify" notes and prevent double-spending attempts of already spent notes.

Reviewer: ```suggestion
**Description**: Based on a given list of nullifiers, returns only the nullifiers that already exist in the contract, i.e. the nullifiers of the notes that have been spent already. Nullifiers "nullify" notes and prevent double-spending attempts of already spent notes.
```

### `leaves_from_height`

**Description**: Feeds the host with all leaves in the tree starting from the given block height.

```rust
pub fn leaves_from_height(&self, height: u64)
```

### `leaves_from_pos`

**Description**: Feeds the host with all leaves in the tree starting from the given tree position.

```rust
pub fn leaves_from_pos(&self, pos: u64)
```

### `sync`

**Description**: Feeds the host with leaves starting from a position, with an optional limit.

```rust
pub fn sync(&self, from: u64, count_limit: u64)
```

### `sync_nullifiers`

**Description**: Feeds the host with nullifiers starting from a position (skipping entries before), with an optional limit.

```rust
pub fn sync_nullifiers(&self, from: u64, count_limit: u64)
```

### `sync_contract_balances`

**Description**: Feeds the host with all contract balances, with an optional limit.

```rust
pub fn sync_contract_balances(&self, from: u64, count_limit: u64)
```

### `sync_accounts`

**Description**: Feeds the host with account data (balances & nonces), with an optional limit.

```rust
pub fn sync_accounts(&self, from: u64, count_limit: u64)
```

### `spend_and_execute`
> *Can only be called from outside the VM*

> **Description**: Main entry point for all transaction executions, handling Phoenix **or** Moonlight transactions. Phoenix & Moonlight are strictly separated, and the spending phase will either go into `spend_phoenix` or `spend_moonlight` based on the transaction type. Afterwards, it executes any contract call included in the transaction. The function manages the complete transaction lifecycle from spending Dusk to executing contract calls.

Reviewer: ```suggestion
**Description**: This function is the main entry point for all transaction executions and manages the complete transaction lifecycle from spending Dusk to executing contract calls and paying gas. As such it is handling both Phoenix **and** Moonlight transactions, even though Phoenix & Moonlight are strictly separated, and the spending phase will either go into `spend_phoenix` or `spend_moonlight` based on the transaction type.
```

```rust
pub fn spend_and_execute(&mut self, tx: Transaction) -> Result<Vec<u8>, ContractError>
```

### `refund`

> *Can only be called from outside the VM*

**Description**: Refunds remaining gas and unclaimed deposits after transaction execution. Emits events with transaction details and refund information.

```rust
pub fn refund(&mut self, gas_spent: u64)
```

### `push_note`

> *Can only be called from outside the VM*

**Description**: Adds a new note to the tree with the specified block height. This is related to the phoenix UTXO model, allowing new outputs to be added.

```rust
pub fn push_note(&mut self, block_height: u64, note: Note) -> Option<Note>
```

### `update_root`

> *Can only be called from outside the VM*

**Description**: Updates the list of tree roots with the current tree root. Enables future verification of notes against historical tree states.

```rust
pub fn update_root(&mut self)
```

### `add_account_balance`

> *Can only be called from outside the VM*

> **Description**: Adds Dusk to an account's balance, creating the account if it doesn't exist.

Reviewer: ```suggestion
**Description**: Adds Dusk to an moonlight account's balance, creating the account if it doesn't exist.
```

```rust
pub fn add_account_balance(&mut self, key: &AccountPublicKey, value: u64)
```

### `sub_account_balance`

> *Can only be called from outside the VM*

> **Description**: Subtracts Dusk from an account's balance if the account exists.

Reviewer: ```suggestion
**Description**: Subtracts Dusk from a moonlight account's balance if the account exists.
```
